### ISSUE TITLE
Carburator Icing

### ISSUE THREAD

**Issue by ghost** [Harald67/c150#42]
Another interesting topic, could we simulate carburator icing?
-usually happens when the air is cold and the power is low
-doesn't require temperatures below 0! (due to lower pressure the aspirator generates)
-coming icing shows through lower RPM while throttle stays the same
-leads to engine failure if carburator is not heated during beginning icing

**Comment by Harald67** [comment]
Yes this can be added of course. But someone will have to do a bit of research.
I saw a graphic with plausible icing condition, was it on the forum ?

**Comment by gilbertohasnofb** [comment]
I just implemented that for the c172p (see branch `bug-360`), and it should get merged in the `master` branch there soon, though probably this enhancement won't make it to FGDATA since I am a bit suspicious of merging it too short time before the release (when we can't test it enough). 

But @onox plans to make a better implementation of this. I am using using a single mathematical formula with temperature, dewpoint and RPM as variables, but he plans to use other atmospheric conditions to make it better reflect RL.

So I would say let's keep this on hold, and I can take a look at it in the future. Is that fine, @Harald67 ?

**Comment by Harald67** [comment]
That's OK with me.
But I'm wondering if we can have something realistic at the end. For example the c150 and the c152 are very similar aircraft but they do not have the same carburetor icing problems, just because the position of the carburetor is different in a c152.

**Comment by gilbertohasnofb** [comment]
> For example the c150 and the c152 are very similar aircraft but they do not have the same carburetor icing problems, just because the position of the carburetor is different in a c152.

Well, you have to keep in mind that the simulation depends on some arbitrary parameters set by the programmer to best simulate something as he sees it. For instance, as much as you can take into consideration the RPM and atmospheric variables, you still need to make some decisions about how this model will end up working: how long does it take for the ice to completely stop the engine? How much RPM is lost when you have ice covering 5% of the carburetor? Etc. So using a same implementation, one can play with the values so that an airplane will not suffer icing at all while the other can suffer a lot. Makes sense?

**Comment by ghost** [comment]
Yep.. So we wait with this till it's implemented in the c172p and then adapt it?

**Comment by gilbertohasnofb** [comment]
Yes. And either we wait a little bit until my implementation is finished, or we wait a bit longer until a better implementation is done by onox. But personally as cool as this feature is I don't think it's a high priority right now. As amazing job as this c150 is already, it still needs a lot of work (e.g. improvements to the 3D model and UV-mapping, which I am anxiously waiting for as what really ticks me is to create textures)

**Comment by ghost** [comment]
in which you are for sure the best ;) (texturing)

**Comment by gilbertohasnofb** [comment]
Thanks, D-ECHO! :smile:

**Comment by ghost** [comment]
Just a general note, I'll be away two weeks from tomorrow on so I can't do any contributions during that time but I'm curious how it'll be when I come back! ;) :D

**Comment by ghost** [comment]
Btw @gilbertohasnofb @Harald67 how to add labels, milestones or assignees?

**Comment by Harald67** [comment]
Perhaps I just need to add you to the project...let's see if that work.

Also note that there is no reason to rush, I know that there are things that need more work but my aim was to have a not so bad version of the aircraft for FG 2013.3 (that should arrives very soon).
Thanks to Gilberto we were able to fix most problems and even add more things that I would have put in that version.

**Comment by ghost** [comment]
yep, works thanks :D

**Comment by ghost** [comment]
and, I know off-topic but I'd again like to **thank you both very much for the beautiful work!!**

**Comment by ghost** [comment]
The C172P seem to recently got a new carburator icing system, is this your implementation (gilberto) or the "better one by onox"? ;)

**Comment by gilbertohasnofb** [comment]
@D-ECHO That was my implementation. We decided to merge it because it seems to work well, but also because we need more people testing for possible bugs which is why we announced it in the forum. So I would give it some time before adapting it here or on other projects, just to make sure everything works as expected.

**Comment by gilbertohasnofb** [comment]
If anyone is interested, here is the discussion about my implementation: https://github.com/Juanvvc/c172p-detailed/pull/826 And here is the full discussion about the merits of certain approaches https://github.com/Juanvvc/c172p-detailed/issues/360

**Comment by ghost** [comment]
Okay, thanks, though I have no expertise on the systems used there, I'll test if it behaves as expected in reality :D

**Comment by gilbertohasnofb** [comment]
>  though I have no expertise on the systems used there, I'll test if it behaves as expected in reality :D

That's exactly what we are looking for! Take a look at the chart in the first link I posted above to see how icing should occur (given a combination of temperature and dew point), then see if it behaves as expected. The first post in that link explains how I did my testing. If you see anything odd, either post in the c172p thread in the forum or in our repo over there :smile:

**Comment by ghost** [comment]
It's now for a pretty long time in the c172p and is working well, if I didn't miss anything.. Would you consider adding it to the c150 now? ;)

**Comment by gilbertohasnofb** [comment]
Well, it has not been a "pretty long time" in my opinion, it has been literally 2 months only :smile: Moreover, I didn't get a single piece of feedback concerning that system up to now, so to be honest there is not a single difference between today and two months ago. The thing is this system in the c172p will be shipped with 2016.4 so many more people will use it, and hopefully I will then receive some feedback (or the lack of it will indicate all is fine). So I ask just a bit more of patience before implementing this here.

**Comment by ghost** [comment]
Okay, that's fine :D

Am 14.11.2016 um 09:24 schrieb Gilberto Agostinho:

**Comment by gilbertohasnofb** [comment]
Thanks for understanding it. Just a little bit longer and then we can import that code here, I just don't want to go through the trouble of fixing the same bugs in two or more repositories

**Comment by ghost** [comment]
Yep I understand, no problem, just didn't want this part to disappear into forgetting

**Comment by gilbertohasnofb** [comment]
@Harald67 if you are interested, I have finished working on the carburettor for the c172p and I think it would be quite easy to adapt that code for the c150.
